src/main/java/org/opentripplanner/updater/vehicle_positions/VehiclePositionPatternMatcher.java

```diff
@@ -0,0 +1,116 @@
+package org.opentripplanner.updater.vehicle_positions;
+
+import com.google.transit.realtime.GtfsRealtime.VehiclePosition;
+import org.opentripplanner.index.model.RealtimeVehiclePosition;
+import org.opentripplanner.model.FeedScopedId;
+import org.opentripplanner.model.Trip;
+import org.opentripplanner.routing.edgetype.TripPattern;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.GraphIndex;
+import org.opentripplanner.routing.vertextype.TransitStop;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Responsible for converting vehicle positions in memory to exportable ones, and associating each position
+ * with a pattern.
+ */
+public class VehiclePositionPatternMatcher {
+    private static final Logger LOG =
+            LoggerFactory.getLogger(VehiclePositionPatternMatcher.class);
+
+    private final GraphIndex graphIndex;
+
+    public VehiclePositionPatternMatcher(Graph graph) {
+        graphIndex = graph.index;
+    }
+
+    /**
+     * Attempts to match each vehicle position to a pattern, then adds each to a pattern
+     * @param vehiclePositions  List of vehicle positions to match to patterns
+     * @param feedId            Feed id of vehicle positions to assist in pattern-matching
+     */
+    public void applyVehiclePositionUpdates(List<VehiclePosition> vehiclePositions, String feedId) {
+        for (VehiclePosition vehiclePosition : vehiclePositions) {
+            if (!vehiclePosition.hasTrip()) {
+                LOG.warn("Realtime vehicle positions without trip IDs are not yet supported.");
+            }
```

Comment:
Should there be a `continue` statement here?

```suggestion
                LOG.warn("Realtime vehicle positions without trip IDs are not yet supported.");
                continue;
            }
```

Reply:
Yes there should be! Only report one error instead of two. Fixed in 1360e72e44678135324c29f7ac4942514fa05270